server.js

```diff
@@ -1,30 +1,102 @@
 import express from 'express'
-import bodyParser from 'body-parser'
 import cors from 'cors'
 
-// If you're using one of our datasets, uncomment the appropriate import below
-// to get started!
-// 
-// import goldenGlobesData from './data/golden-globes.json'
-// import avocadoSalesData from './data/avocado-sales.json'
-// import booksData from './data/books.json'
-// import netflixData from './data/netflix-titles.json'
-// import topMusicData from './data/top-music.json'
-
-// Defines the port the app will run on. Defaults to 8080, but can be 
-// overridden when starting the server. For example:
-//
-//   PORT=9000 npm start
+import netflixTitles from './data/netflix-titles.json'
+
 const port = process.env.PORT || 8080
 const app = express()
 
-// Add middlewares to enable cors and json body parsing
 app.use(cors())
-app.use(bodyParser.json())
+app.use(express.json())
 
-// Start defining your routes here
 app.get('/', (req, res) => {
-  res.send('Hello world')
+  res.send('Hi! This is the root-endpoint for my Netflix API.')
+})
+
+// endpoint for all titles + querys
+app.get('/titles', (req, res) => {
+  const { released, name, country, category, rating } = req.query
+  let titles = netflixTitles
+
+  if (released) {
+    titles = titles
+      .filter(title => title.release_year === +released)
+  }
+
+  if (name) {
+    titles = titles
+      .filter(title => title.title.toString().toLowerCase().replace(/\s/g, '').includes(name.toLowerCase().replace(/\s/g, '')))
+  }
+
+  if (country) {
+    titles = titles
+      .filter(title => title.country.toLowerCase().replace(/\s/g, '').includes(country.toLowerCase().replace(/\s/g, '')))
+  }
+
+  if (category) {
+    titles = titles
+      .filter(title => title.listed_in.toLowerCase().replace(/\s/g, '').includes(category.toLowerCase().replace(/\s/g, '')))
+  }
+
+  if (rating) {
+    titles = titles
+      .filter(title => title.rating.toLowerCase().replace(/-/g, '') === rating.toLowerCase().replace(/-/g, ''))
+  }
+
+  if (titles.length) {
+    res.json({ length: titles.length, data: titles })
+  } else {
+    res.status(404).send({ message: `No match!` })
+  }
+
+})
+
+//param to get a single title
+app.get('/titles/:id', (req, res) => {
+  const { id } = req.params
+  const titleId = netflixTitles.find((title) => title.show_id === +id)
+
+  if (!titleId) {
+    res.status(404).send({ error: `No title with id ${id} found.` })
```

Comment:
you repeating this structure a lot, so it would be nice to create some errorHandler function and call it when needed 🤓